### PR TITLE
Fix Coverity 1498612 & 1503221: integer overflow

### PR DESCRIPTION
**Pull request by paulidale**
Both are the same issue and both as false positives.  Annotate the line so that this is ignored.

- [ ] documentation is added or updated
- [ ] tests are added or updated

### CODE REVIEW

File: crypto/ec/curve448/curve448.c [hunk 1]
```diff
@@ -586,9 +586,15 @@ static int recode_wnaf(struct smvt_control *control,
             int32_t delta = odd & mask;
 
             assert(position >= 0);
-            assert(pos < 32);       /* can't fail since current & 0xFFFF != 0 */
             if (odd & (1 << (table_bits + 1)))
                 delta -= (1 << (table_bits + 1));
+            /*
+             * Coverity gets confuxed by the value os pos, thining it might be
```
Comment (bernd-edlinger) [inline crypto/ec/curve448/curve448.c:592]: 2 typos here: confused + thining

Comment (paulidale) [reply crypto/ec/curve448/curve448.c:592]: and the third: os

Comment (paulidale) [reply crypto/ec/curve448/curve448.c:592]: fixed now

File: crypto/ec/curve448/curve448.c [hunk 2]
```diff
@@ -586,9 +586,15 @@ static int recode_wnaf(struct smvt_control *control,
             int32_t delta = odd & mask;
 
             assert(position >= 0);
-            assert(pos < 32);       /* can't fail since current & 0xFFFF != 0 */
             if (odd & (1 << (table_bits + 1)))
                 delta -= (1 << (table_bits + 1));
+            /*
+             * Coverity gets confused by the value of pos, thinking it might be
+             * 32. This would require current & 0xFFFF to be zero which isn't
```
Comment (bernd-edlinger) [inline crypto/ec/curve448/curve448.c:593]: You might use 2 spaces after the period here, like it is done in the following line.
Well its GNU style, so you don't have to do it....

Comment (paulidale) [reply crypto/ec/curve448/curve448.c]: We're not at all consistent about spaces after full stops.
I've added the space.

**Review by bernd-edlinger — APPROVED** [review]

**Review by t8m — APPROVED** [review]

**Comment by paulidale** [comment]
Dropping 1.1.1 and 3.0 branches unless somebody really wants them.

**Comment by paulidale** [comment]
Merged to master only.  Thanks for the reviews.

**Pull request closed without merging**